src/main/java/org/myrobotlab/sensor/TimeEncoder.java

```diff
@@ -79,8 +85,94 @@
 
   boolean enabled = true;
 
+  static class Positions implements Runnable {
+
+    static Positions instance = null;
+    Map<String, Double> positions = new HashMap<>();
```

Comment:
I'm a bit worried about threading here... perhaps we should be using a ConcurrentHashMap instead?